fastly/digitalocean_test.go

```diff
@@ -126,65 +201,108 @@ func TestClient_DigitalOceans(t *testing.T) {
+		t.Errorf("bad format_version: %q", digitaloceanCreateResp1.FormatVersion)
 	}
-	if digitalocean.TimestampFormat != ndigitalocean.TimestampFormat {
-		t.Errorf("bad timestamp_format: %q", digitalocean.TimestampFormat)
+	if digitaloceanCreateResp1.TimestampFormat != digitaloceanGetResp.TimestampFormat {
+		t.Errorf("bad timestamp_format: %q", digitaloceanCreateResp1.TimestampFormat)
 	}
-	if digitalocean.Placement != ndigitalocean.Placement {
-		t.Errorf("bad placement: %q", digitalocean.Placement)
+	if digitaloceanCreateResp1.Placement != digitaloceanGetResp.Placement {
+		t.Errorf("bad placement: %q", digitaloceanCreateResp1.Placement)
 	}
-	if digitalocean.PublicKey != ndigitalocean.PublicKey {
-		t.Errorf("bad public_key: %q", digitalocean.PublicKey)
+	if digitaloceanCreateResp1.PublicKey != digitaloceanGetResp.PublicKey {
+		t.Errorf("bad public_key: %q", digitaloceanCreateResp1.PublicKey)
+	}
+	if digitaloceanCreateResp1.CompressionCodec != digitaloceanGetResp.CompressionCodec {
+		t.Errorf("bad compression_codec: %q", digitaloceanCreateResp1.CompressionCodec)
 	}
 
 	// Update
-	var udigitalocean *DigitalOcean
+	var digitaloceanUpdateResp1, digitaloceanUpdateResp2, digitaloceanUpdateResp3 *DigitalOcean
 	record(t, "digitaloceans/update", func(c *Client) {
-		udigitalocean, err = c.UpdateDigitalOcean(&UpdateDigitalOceanInput{
+		digitaloceanUpdateResp1, err = c.UpdateDigitalOcean(&UpdateDigitalOceanInput{
+			ServiceID:        testServiceID,
+			ServiceVersion:   tv.Number,
+			Name:             "test-digitalocean",
+			NewName:          String("new-test-digitalocean"),
+			Domain:           String("nyc3.digitaloceanspaces.com"),
+			CompressionCodec: String("zstd"),
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	record(t, "digitaloceans/update2", func(c *Client) {
+		digitaloceanUpdateResp2, err = c.UpdateDigitalOcean(&UpdateDigitalOceanInput{
+			ServiceID:        testServiceID,
+			ServiceVersion:   tv.Number,
+			Name:             "test-digitalocean-2",
+			CompressionCodec: String("zstd"),
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	record(t, "digitaloceans/update3", func(c *Client) {
+		digitaloceanUpdateResp3, err = c.UpdateDigitalOcean(&UpdateDigitalOceanInput{
 			ServiceID:      testServiceID,
 			ServiceVersion: tv.Number,
-			Name:           "test-digitalocean",
-			NewName:        String("new-test-digitalocean"),
-			Domain:         String("nyc3.digitaloceanspaces.com"),
+			Name:           "test-digitalocean-3",
+			GzipLevel:      Uint(9),
 		})
 	})
 	if err != nil {
 		t.Fatal(err)
 	}
-	if udigitalocean.Name != "new-test-digitalocean" {
-		t.Errorf("bad name: %q", udigitalocean.Name)
+
+	if digitaloceanUpdateResp1.Name != "new-test-digitalocean" {
+		t.Errorf("bad name: %q", digitaloceanUpdateResp1.Name)
+	}
+	if digitaloceanUpdateResp1.Domain != "nyc3.digitaloceanspaces.com" {
+		t.Errorf("bad domain: %q", digitaloceanUpdateResp1.Domain)
+	}
```

Comment:
Should we not also check:

```go
if digitaloceanUpdateResp1.CompressionCodec != "zstd" {
```

as it was set to snappy originally and now we're setting it to zstd.

Reply:
Good call, I added this check.